userspace/libsinsp/sinsp_filtercheck_event.cpp

```diff
@@ -1198,18 +1198,17 @@ uint8_t* sinsp_filter_check_event::extract_single(sinsp_evt* evt,
 
 		ASSERT(m_inspector != NULL);
 
+		sinsp_evt::param_fmt format =
+		        m_is_compare ? sinsp_evt::param_fmt::PF_NORMAL : m_inspector->get_buffer_format();
+
 		if(m_argid != -1) {
 			if(m_argid >= (int32_t)evt->get_num_params()) {
 				return NULL;
 			}
 
-			argstr = evt->get_param_as_str(m_argid,
-			                               &resolved_argstr,
-			                               m_inspector->get_buffer_format());
+			argstr = evt->get_param_as_str(m_argid, &resolved_argstr, format);
```

Comment:
Question: is this the only place where the fix applies? What about embedding it inside `evt->get_param_as_str()` by passing the additional `m_is_compare` boolean? This would enforce any other place to choose the right logic (ie: is a compare? PF_NORMAL, else use the buffer format); or perhaps even better inside:
```
m_inspector->get_buffer_format(bool is_compare)
```

Reply:
I think it's currently clear what the function does: `get_param_as_str()` should do one thing, formatting one param as string. The format parameter tells it how it is supposed to be formatted. There may be other things that may affect how that parameter is formatted which are not necessarily the `is_compare` flag, for instance, the function is used to format some fields during syscall parsing, testing or other things. Perhaps in some other comparisons we would like to use a different format.

Reply:
Thanks for the explanation!